psi-notify.c

```diff
@@ -249,7 +249,7 @@ static void config_reset_user_facing(void) {
 #define WATCHDOG_GRACE_PERIOD_SEC 5
 #define SEC_TO_USEC 1000000
 static void watchdog_update_usec(void) {
-    sd_notifyf(0, "WATCHDOG_USEC=%d",
+    sd_notifyf(0, "WATCHDOG_USEC=%ld",
```

Comment:
This is liable to break, since you don't know portably how wide `time_t` is. Instead, cast the result to `unsigned long` first.

Reply:
Hmm, let's see if long long works on 32-bit ARM and i686. Will test.

Reply:
long is even fine if you have problems, to be honest.